lib/test.js

```diff
@@ -92,8 +93,17 @@ Test.prototype.run = function () {
         this.timeoutAfter(this._timeout);
     }
     this.emit('prerun');
-    this._cb(this);
+    var callbackReturn = this._cb(this);
     this.emit('run');
+
+    if (typeof Promise !== 'undefined') {
```

Comment:
let's ensure Promise is a function, as is Promise.resolve

Reply:
done. Also changed in lib/test.js